src/cpp/rtps/builtin/liveliness/WLP.cpp

```diff
@@ -77,17 +98,42 @@ WLP::~WLP()
     delete(this->mp_builtinReaderHistory);
     delete(this->mp_builtinWriterHistory);
     delete(this->mp_listener);
-    if(this->mp_livelinessAutomatic!=nullptr)
-        delete(mp_livelinessAutomatic);
-    if(this->mp_livelinessManRTPSParticipant!=nullptr)
-        delete(this->mp_livelinessManRTPSParticipant);
+
+    delete pub_liveliness_manager_;
+    delete sub_liveliness_manager_;
 }
 
 bool WLP::initWL(RTPSParticipantImpl* p)
 {
-    logInfo(RTPS_LIVELINESS,"Beginning Liveliness Protocol");
+    logInfo(RTPS_LIVELINESS,"Initializing Liveliness Protocol");
+
     mp_participant = p;
 
+    pub_liveliness_manager_ = new LivelinessManager(
+                [&](const GUID_t& guid,
+                    const LivelinessQosPolicyKind& kind,
+                    const Duration_t& lease_duration) -> void
+                {
+                    pub_liveliness_lost(guid, kind, lease_duration);
```

Comment:
I was initially doing that, but then I saw during the revision of https://github.com/eProsima/Fast-RTPS/pull/552 that @richiprosima tends to do it this way (see changes to `PublisherImpl.cpp`). The reason not to add the implementation inside the lambda is because I thought it would be too long, specially for `pub_liveliness_lost()`

Reply:
Another style discussion here :). What should we do with large lambda functions?

Reply:
As agreed, it is better to use lambdas in this case, and for clarity, we will keep the call to the method inside the lambda.